src/Uno.UI/UI/Xaml/UIElement.Layout.crossruntime.cs

```diff
@@ -235,14 +243,29 @@ private void DoMeasure(Size availableSize)
 				if (isDirty)
 				{
 					// We must reset the flag **BEFORE** doing the actual measure, so the elements are able to re-invalidate themselves
-					ClearLayoutFlags(LayoutFlag.MeasureDirty | LayoutFlag.MeasureDirtyPath);
+					if (partOfLayoutCycle)
+					{
+						ClearLayoutFlags(LayoutFlag.MeasureDirty | LayoutFlag.MeasureDirtyPath);
+					}
 
 					// The dirty flag is explicitly set on this element
 #if DEBUG
 					try
 #endif
 					{
+						var disposables = new List<IDisposable>();
+						// This checks makes sure that this is part of the layout cycle,
+						// and not just a random call to a parent's Measure()
+						if (partOfLayoutCycle)
+						{
+							foreach (var child in GetChildren())
```

Comment:
I'm really worried about the perf impact of this: Creating a new list on each measure, adding an enumeration of the `Children` collection and finally enumerating the newly created collection ... I would rather check in the `child.Measure` if the direct parent is `CurrentlyBeingMeasuredByParentInLayoutCycle`.